--- origsrc/redland-1.0.8/librdf/Makefile.am	2008-07-04 01:56:25.000000000 -0500
+++ src/redland-1.0.8/librdf/Makefile.am	2008-07-21 00:59:01.656250000 -0500
@@ -108,32 +108,32 @@
 plugin_LTLIBRARIES += librdf_storage_tstore.la
 librdf_storage_tstore_la_SOURCES = rdf_storage_tstore.c
 librdf_storage_tstore_la_CPPFLAGS = @TSTORE_CPPFLAGS@
-librdf_storage_tstore_la_LIBADD = @TSTORE_LIBS@
-librdf_storage_tstore_la_LDFLAGS = -module -avoid-version
+librdf_storage_tstore_la_LIBADD = @TSTORE_LIBS@ librdf.la
+librdf_storage_tstore_la_LDFLAGS = -module -avoid-version -no-undefined
 endif
 
 if STORAGE_MYSQL
 plugin_LTLIBRARIES += librdf_storage_mysql.la
 librdf_storage_mysql_la_SOURCES = rdf_storage_mysql.c
 librdf_storage_mysql_la_CPPFLAGS = @MYSQL_CPPFLAGS@
-librdf_storage_mysql_la_LIBADD = @MYSQL_LIBS@
-librdf_storage_mysql_la_LDFLAGS = -module -avoid-version
+librdf_storage_mysql_la_LIBADD = @MYSQL_LIBS@ librdf.la
+librdf_storage_mysql_la_LDFLAGS = -module -avoid-version -no-undefined
 endif
 
 if STORAGE_SQLITE
 plugin_LTLIBRARIES += librdf_storage_sqlite.la
 librdf_storage_sqlite_la_SOURCES = rdf_storage_sqlite.c
 librdf_storage_sqlite_la_CPPFLAGS = @SQLITE_CPPFLAGS@
-librdf_storage_sqlite_la_LIBADD = @SQLITE_LIBS@
-librdf_storage_sqlite_la_LDFLAGS = -module -avoid-version
+librdf_storage_sqlite_la_LIBADD = @SQLITE_LIBS@ librdf.la
+librdf_storage_sqlite_la_LDFLAGS = -module -avoid-version -no-undefined
 endif
 
 if STORAGE_POSTGRESQL
 plugin_LTLIBRARIES += librdf_storage_postgresql.la
 librdf_storage_postgresql_la_SOURCES = rdf_storage_postgresql.c
 librdf_storage_postgresql_la_CPPFLAGS = @POSTGRESQL_CPPFLAGS@
-librdf_storage_postgresql_la_LIBADD = @POSTGRESQL_LIBS@
-librdf_storage_postgresql_la_LDFLAGS = -module -avoid-version
+librdf_storage_postgresql_la_LIBADD = @POSTGRESQL_LIBS@ librdf.la
+librdf_storage_postgresql_la_LDFLAGS = -module -avoid-version -no-undefined
 endif
 
 else
@@ -175,7 +175,7 @@
 EXTRA_DIST=\
 redland.spec redland.spec.in
 
-librdf_la_LDFLAGS = -version-info @LIBRDF_LIBTOOL_VERSION@ \
+librdf_la_LDFLAGS = -version-info @LIBRDF_LIBTOOL_VERSION@ -no-undefined \
 @LIBRDF_LDFLAGS@ @LIBRDF_EXTERNAL_LIBS@ $(MEM_LIBS)
 
 pkgdata_DATA=
